airbyte-workers/src/main/java/io/airbyte/workers/DefaultSyncWorker.java

```diff
@@ -83,9 +84,13 @@ public DefaultSyncWorker(
   public StandardSyncOutput run(StandardSyncInput syncInput, Path jobRoot) throws WorkerException {
     long startTime = System.currentTimeMillis();
 
-    LOGGER.info("configured sync modes: {}", syncInput.getCatalog().getStreams()
+    LOGGER.info("configured source sync modes: {}", syncInput.getCatalog().getStreams()
         .stream()
         .collect(Collectors.toMap(s -> s.getStream().getName(), s -> s.getSyncMode() != null ? s.getSyncMode() : SyncMode.FULL_REFRESH)));
+    LOGGER.info("configured destination sync modes: {}", syncInput.getCatalog().getStreams()
+        .stream()
+        .collect(Collectors.toMap(s -> s.getStream().getName(),
+            s -> s.getDestinationSyncMode() != null ? s.getDestinationSyncMode() : DestinationSyncMode.APPEND)));
```

Comment:
is this assumption safe? it seems like if syncMode is full_refresh and destinationSyncMode is null then we'll end up with destinationSyncMode being set to append. In the case i think the intended configuration is full refresh, but we'll end up with incremental. it seems like for now if destinationSyncMode is empty you need to set using information from sync mode to figure out what it should be? i could be misunderstanding. 

maybe you could write a test that shows the mapping of the possible configurations to what you expect them to be so that we can all be on the same page?

Reply:
Discussed offline and agreed on moving forward with 2 changes:
1. we'll be removing the default values of sync modes and raising exceptions if they are not defined...
1. I will be providing a Migration Script to make sure destination sync modes are defined when upgrading versions and are keeping the same behavior as before the upgrade

Reply:
Since they become required I'll start another PR for that change instead.
In the meantime, I can merge this one as it is